scripts/prepare.js

```diff
@@ -38,8 +38,12 @@ function logError(type, packageJson) {
 const packageJson = getPackageJson();
 
 removeDist();
-babelify({ errorCallback: () => logError('js', packageJson) });
-removeTsFromDist();
-tscfy({ errorCallback: () => logError('ts', packageJson) });
+if (packageJson && packageJson.types && packageJson.types.indexOf('d.ts') !== -1) {
+  tscfy({ errorCallback: () => logError('ts', packageJson) });
+} else {
+  babelify({ errorCallback: () => logError('js', packageJson) });
+  removeTsFromDist();
+  tscfy({ errorCallback: () => logError('ts', packageJson) });
+}
```

Comment:
Pure ts projects may cause unexpected behavior when it runs through babelify as well
In my case babelify copied `tsx` files into dist